### PR TITLE
[Do not merge] Attach all local listeners as captured events

### PR DESCRIPTION
**Pull request by nhunzaker**
I noticed in https://github.com/facebook/react/pull/9333 that events that don't bubble are still captureable (scroll, wheel).  So that got me curious about all of the other events we attach locally for that very reason.

I've used the following test page to confirm this:
https://codepen.io/nhunzaker/pen/babXLO

https://codepen.io/anon/pen/YYKoJW?editors=1010

**What I need help with**

Is this even a good idea? 😛. Are there any design issues with using captured events this way? 

Right now, this PR only moves the local attachment over, as needed, to ReactBrowserEventEmitter. There is a more aggressive change: **what if we attached all events as capture?** Period. Given bubbling is synthetic anyway, what could go wrong if we attached everything using capture? Then we wouldn't really need to distinguish local listeners other the handful we have for focus and scrolling.

**Things that still need confirmation:**

- [x] iframe: onload
- [x] object: onload
- [ ] video,audio: abort, canplay, canplaythrough, durationchange, emptied, encrypted, ended, error, loadeddata, loadedmetadata, loadstart, pause, play, playing, progress, ratechange, seeked, seeking, stalled, suspend, timeupdate, volumechange, waiting
- [ ] source: error
- [x] img ([test](https://codepen.io/anon/pen/YYKoJW?editors=1010)): onload, onerror
- [x] image (svg): onload, onerror
- [x] form: reset, submit
- [x] details: toggle
- [x] input: invalid
- [x] select: invalid
- [x] textarea: invalid

So basically media events are left. For all the others, I've tested:

* Chrome 43, 62
* Firefox 42, 58
* IE 9, IE 10, IE 11
** No SVG Load event
** Details are not supported
* Safari 11, 7.1
** Details does not dispatch until Safari 10.1 - https://codepen.io/nhunzaker/pen/qpBWqo
* Edge 14, 15, 16
** No SVG Load event - https://github.com/facebook/react/issues/11058
** Details are not supported
* Android UTC on 4.4
** Details does not dispatch until Android 5 - https://codepen.io/nhunzaker/pen/qpBWqo
* iOS Safari 7, 10
** Details does not dispatch until Safari 10.1 - https://codepen.io/nhunzaker/pen/qpBWqo

### CODE REVIEW

**Comment by nhunzaker** [comment]
I learned what I needed from this. Closing.
